### PR TITLE
Ability to provide custom NgModuleRef

### PR DESCRIPTION
**Pull request by msgoloborodov**

### CODE REVIEW

File: projects/ng-dynamic-component/src/lib/dynamic.component.ts
Comment (gund) [inline projects/ng-dynamic-component/src/lib/dynamic.component.ts:74]: I think it makes sense to also add support for the `environmentInjector` as it's going to replace `ngModuleRef` some time in the future (as per [Angular docs](https://angular.io/api/core/ViewContainerRef#createcomponent): "This [environmentInjector] option is intended to replace the ngModuleRef parameter.")?

File: projects/ng-dynamic-component/src/lib/dynamic.component.ts
Comment (gund) [inline projects/ng-dynamic-component/src/lib/dynamic.component.ts:46]:
```suggestion
  ndcDynamicNgModule?: NgModuleRef<unknown>;
```

**Review by gund — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could you please also add [unit tests](https://github.com/gund/ng-dynamic-component/blob/master/projects/ng-dynamic-component/src/lib/dynamic.component.spec.ts) for this new options?
To test `ngModuleRef` you could provide something in that module and then try to inject that in the injected component's injector to see if it's available.
Same strategy could be used for `environmentInjector` option as well.

P.S. You can use Github Codespaces to make changes fully online in full vscode ide and run build/tests there without having to install anything locally:
image

File: projects/ng-dynamic-component/src/lib/dynamic.component.ts
Comment (gund) [reply projects/ng-dynamic-component/src/lib/dynamic.component.ts:46]: Also you forgot to import `NgModuleRef` and [the build has failed](https://ci.appveyor.com/project/gund/ng-dynamic-component/builds/46014992#L43).

Comment (msgoloborodov) [reply projects/ng-dynamic-component/src/lib/dynamic.component.ts]: fixed

File: projects/ng-dynamic-component/src/lib/dynamic.component.ts
Comment (gund) [inline projects/ng-dynamic-component/src/lib/dynamic.component.ts:51]:
```suggestion
  ndcDynamicNgModuleRef?: NgModuleRef<unknown>;
  @Input()
  ndcDynamicEnvironmentInjector?: EnvironmentInjector | NgModuleRef<unknown>;
```

File: projects/ng-dynamic-component/src/lib/dynamic.component.spec.ts
Comment (gund) [inline projects/ng-dynamic-component/src/lib/dynamic.component.spec.ts:40]:
```suggestion
    ngModuleRef?: NgModuleRef<unknown>;
    environmentInjector?: EnvironmentInjector | NgModuleRef<unknown>;
```

**Review by gund — CHANGES_REQUESTED** [review]
Looks good!

Before merging I'd like to make a few small adjustments to match the types with Angular and also add the following text to the readme.md file as a documentation [after this location](https://github.com/gund/ng-dynamic-component/blob/5df76ff9cb2f7a3c52a44e93a9e9bac9e9c2c9ce/README.md?plain=1#L419):

```md
**Since v10.6.0**: You can provide custom NgModuleRef (`[ndcDynamicNgModuleRef]`)
or EnvironmentInjector (`[ndcDynamicEnvironmentInjector]`) for your dynamic component.
```

Alternatively I can make those changes myself but you would have to [allow maintainers to push to this PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests).

File: projects/ng-dynamic-component/src/lib/dynamic.component.ts
Comment (msgoloborodov) [reply projects/ng-dynamic-component/src/lib/dynamic.component.ts:51]: Option "Allow edits from maintainers" is checked. Also I added you as a collaborator to my fork.

**Pull request merged**
